File: src/main/java/no/nav/sosialhjelp/soknad/business/db/repositories/soknadmetadata/SoknadMetadataRepository.java
package no.nav.sosialhjelp.soknad.business.db.repositories.soknadmetadata;

import no.nav.sosialhjelp.soknad.business.domain.SoknadMetadata;

import java.time.LocalDateTime;
import java.util.List;

public interface SoknadMetadataRepository {

    Long hentNesteId();
    void opprett(SoknadMetadata metadata);
    void oppdater(SoknadMetadata metadata);
    SoknadMetadata hent(String behandlingsId);
    List<SoknadMetadata> hentBehandlingskjede(String behandlingsId);
    int hentAntallInnsendteSoknaderEtterTidspunkt(String fnr, LocalDateTime tidspunkt);
    List<SoknadMetadata> hentSvarUtInnsendteSoknaderForBruker(String fnr);
    List<SoknadMetadata> hentAlleInnsendteSoknaderForBruker(String fnr);
    List<SoknadMetadata> hentPabegynteSoknaderForBruker(String fnr);
    List<SoknadMetadata> hentPabegynteSoknaderForBruker(String fnr, boolean lestDittNav);
    List<SoknadMetadata> hentInnsendteSoknaderForBrukerEtterTidspunkt(String fnr, LocalDateTime after);
    void oppdaterLestDittNav(SoknadMetadata soknadMetadata, String fnr);
}

File: src/main/kotlin/no/nav/sosialhjelp/soknad/db/repositories/soknadmetadata/SoknadMetadataRepository.kt
package no.nav.sosialhjelp.soknad.db.repositories.soknadmetadata

import java.time.LocalDateTime

interface SoknadMetadataRepository {
    fun hentNesteId(): Long

    fun opprett(metadata: SoknadMetadata)

    fun oppdater(metadata: SoknadMetadata?)

    fun hent(behandlingsId: String?): SoknadMetadata?

    fun hentAntallInnsendteSoknaderEtterTidspunkt(
        fnr: String?,
        tidspunkt: LocalDateTime?,
    ): Int?

    fun hentAlleInnsendteSoknaderForBruker(fnr: String): List<SoknadMetadata>

    fun hentPabegynteSoknaderForBruker(fnr: String): List<SoknadMetadata>

    fun hentPabegynteSoknaderForBruker(
        fnr: String,
        lest: Boolean,
    ): List<SoknadMetadata>

    fun hentInnsendteSoknaderForBrukerEtterTidspunkt(
        fnr: String,
        tidsgrense: LocalDateTime,
    ): List<SoknadMetadata>

    fun oppdaterLest(
        soknadMetadata: SoknadMetadata,
        fnr: String,
    )
}
